pkg/config/validator.go

```diff
@@ -307,7 +312,48 @@ func validateSourceBindings(sl validator.StructLevel, sources map[string]Sources
 		}
 	}
 
-	validateBindPlugins(sl, enabledPluginsViaBindings)
+	validateBoundPlugins(sl, enabledPluginsViaBindings)
+	validatePluginRBAC(sl, sources, bindings)
+}
+
+func validatePluginRBAC[P providesPlugins](sl validator.StructLevel, hasPlugins map[string]P, bindings []string) {
```

Comment:
I'd recommend renaming `hasPlugins` to something else (I know you probably won't like `pluginMap` 😄  - `pluginConfigs`?) `HasXXX` pattern is usually used for bool variables, which might be confusing.